<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the TBSignatureProfiler • TBSignatureProfiler</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Introduction to the TBSignatureProfiler">
<meta property="og:description" content="TBSignatureProfiler">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">TBSignatureProfiler</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li>
  <a href="../../articles/rmd/TBSig_Vignette.html">Command Line Analysis</a>
</li>
<li>
  <a href="../../articles/rmd/OriginalModelTutorial.html">Original Models</a>
</li>
<li>
  <a href="../../articles/rmd/shinytutorial.html">Interactive Shiny Analysis</a>
</li>
<li>
  <a href="../../articles/rmd/SignatureAddition.html">Adding Signatures to the Profiler</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li>
  <a href="../../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/TBSignatureProfiler">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TBSig_Vignette_files/header-attrs-2.9/header-attrs.js"></script><script src="TBSig_Vignette_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="TBSig_Vignette_files/jquery-3.5.1/jquery.min.js"></script><link href="TBSig_Vignette_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="TBSig_Vignette_files/datatables-binding-0.18/datatables.js"></script><link href="TBSig_Vignette_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="TBSig_Vignette_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="TBSig_Vignette_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="TBSig_Vignette_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="TBSig_Vignette_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the TBSignatureProfiler</h1>
                        <h4 class="author">Aubrey Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                              <h4 class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wej@bu.edu" class="email">wej@bu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/TBSignatureProfiler/blob/master/vignettes/rmd/TBSig_Vignette.Rmd"><code>vignettes/rmd/TBSig_Vignette.Rmd</code></a></small>
      <div class="hidden name"><code>TBSig_Vignette.Rmd</code></div>

    </div>

    
    
<div id="introduction-to-the-tbsignatureprofiler" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction-to-the-tbsignatureprofiler" class="anchor"></a>Introduction to the TBSignatureProfiler</h1>
<p>Tuberculosis (TB) is the leading cause of infectious disease mortality worldwide, causing on average nearly 1.4 million deaths per year. A consistent issue faced in controlling TB outbreak is difficulty in diagnosing individuals with particular types of TB infections for which bacteria tests (e.g., via GeneXpert, sputum) prove inaccurate. As an alternative mechanism of diagnosis for these infections, researchers have discovered and published multiple gene expression signatures as blood-based disease biomarkers. In this context, gene signatures are defined as a combined group of genes with a uniquely characteristic pattern of gene expression that occurs as a result of a medical condition. To date, more than 30 signatures have been published by researchers, though most have relatively low cross-condition validation (e.g., testing TB in samples from diverse geographic and comorbidity backgrounds). Furthermore, these signatures have never been formally collected and made available as a single unified resource.</p>
<p>We aim to provide the scientific community with a resource to access these aggregated signatures and to create an efficient means for their visual and quantitative comparison via open source software. This necessitated the development of the TBSignatureProfiler, a novel R package which delivers a computational profiling platform for researchers to characterize the diagnostic ability of existing signatures in multiple comorbidity settings. This software allows for signature strength estimation via several enrichment methods and subsequent visualization of single- and multi-pathway results. Its signature evaluation functionalities include signature profiling, AUC bootstrapping, and leave-one-out cross-validation (LOOCV) of logistic regression to approximate TB samples’ status. Its plotting functionalities include sample-signature score heatmaps, bootstrap AUC and LOOCV boxplots, and tables for presenting results.</p>
<p>More recently, the TBSignatureProfiler has undertaken a new role in analyzing signatures across multiple chronic airway diseases, the most recent being COVID-19 (see the <code>COVIDsignatures</code> object). As we grow and expand the TBSignatureProfiler, we hope to add signatures from multiple diseases to improve the package’s utility in the area of gene signature comparison.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>In order to install the TBSignatureProfiler from Bioconductor, run the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"TBSignatureProfiler"</span><span class="op">)</span></code></pre></div>
</div>
<div id="compatibility-with-summarizedexperiment-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#compatibility-with-summarizedexperiment-objects" class="anchor"></a>Compatibility with SummarizedExperiment objects</h1>
<p>While the TBSignatureProfiler often allows for the form of a <code>data.frame</code> or <code>matrix</code> as input data, the most ideal form of input for this package is that of the <code>SummarizedExperiment</code> object. This is an amazing data structure that is being developed by the  as part of the <em><a href="https://bioconductor.org/packages/3.13/SummarizedExperiment">SummarizedExperiment</a></em> package. It is able to store data matrices along with annotation information, metadata, and reduced dimensionality data (PCA, t-SNE, etc.). To learn more about proper usage and context of the <code>SummarizedExperiment</code> object, you may want to take a look at the <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">package vignette</a>. A basic understanding of the <code>assay</code> and <code>colData</code> properties of a <code>SummarizedExperiment</code> will be useful for the purposes of this vignette.</p>
<p>To install the <code>SummarizedExperiment</code> package, run the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"SummarizedExperiment"</span><span class="op">)</span></code></pre></div>
</div>
<div id="a-tutorial-for-the-tbsignatureprofiler" class="section level1">
<h1 class="hasAnchor">
<a href="#a-tutorial-for-the-tbsignatureprofiler" class="anchor"></a>A Tutorial for the TBSignatureProfiler</h1>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/compbiomed/TBSignatureProfiler">TBSignatureProfiler</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-dataset-from-a-summarizedexperiment-object" class="section level2">
<h2 class="hasAnchor">
<a href="#load-dataset-from-a-summarizedexperiment-object" class="anchor"></a>Load dataset from a SummarizedExperiment object</h2>
<p>In this tutorial, we will work with HIV and Tuberculosis (TB) gene expression data in a <code>SummarizedExperiment</code> format. This dataset is included in the TBSignatureProfiler package and can be loaded into the global environment with <code><a href="https://rdrr.io/r/utils/data.html">data("TB_hiv")</a></code>. The 31 samples in the dataset are marked as either having both TB and HIV infection, or HIV infection only.</p>
<p>We begin by examining the dataset, which contains a matrix of counts information (an “assay” in SummarizedExperiment terms) and another matrix of meta data information on our samples (the “colData”). We will also generate a few additional assays; these are the log(counts), the counts per million (CPM) reads mapped, and the log(CPM) assays.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## HIV/TB gene expression data, included in the package</span>
<span class="va">hivtb_data</span> <span class="op">&lt;-</span> <span class="va">TB_hiv</span>

<span class="co">### Note that we have 25,369 genes, 33 samples, and 1 assay of counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 25369    31</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We start with only one assay</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 1
## names(1): counts</code></pre>
<p>We now make a log counts, CPM and log CPM assay.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Make a log counts, CPM and log CPM assay</span>
<span class="va">hivtb_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/mkAssay.html">mkAssay</a></span><span class="op">(</span><span class="va">hivtb_data</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, counts_to_CPM <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co">### Check to see that we now have 4 assays</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></code></pre></div>
<pre><code>## List of length 4
## names(4): counts cpm log_counts log_cpm</code></pre>
</div>
<div id="profile-the-data" class="section level2 tabset">
<h2 class="hasAnchor">
<a href="#profile-the-data" class="anchor"></a>Profile the data</h2>
<p>The TBSignatureProfiler enables comparison of multiple Tuberculosis gene signatures. The package currently contains information on 34 signatures for comparison. The default signature list object for most functions here is <code>TBsignatures</code>, although a list with publication-given signature names is also available as <code>TBcommon</code>. Data frames of annotation information for these signatures, including information on associated disease and tissue type, can be accessed as <code>sigAnnotData</code> and <code>common_sigAnnotData</code> respectively.</p>
<p>With the <code>runTBSigProfiler</code> function, we are able to score these signatures with a selection of algorithms, including <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7">gene set variation analysis</a> (GSVA) (Hänzelmann et al, 2013), <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7">single-sample GSEA</a> (ssGSEA) (Barbie et al, 2009), and the <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7">ASSIGN</a> pathway profiling toolkit (Shen et al, 2015). For a complete list of included scoring methods, run <code><a href="../../reference/runTBsigProfiler.html">?runTBsigProfiler</a></code> in the terminal.</p>
<p>Here, we evaluate all signatures included in the package with ssGSEA. Paraphrasing from the <a href="http://software.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection/4">ssGSEA documentation</a>, for each pairing of one of the 31 samples and its gene set, ssGSEA calculates a separate enrichment score independent of the phenotypic labeling (in this case, whether a sample has HIV/TB, or HIV only). The single sample’s gene expression profile is then transformed to a gene set enrichment profile. A score from the set profile represents the activity level of the biological process in which the gene set’s members are coordinately up- or down-regulated.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## List all signatures in the profiler</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"TBsignatures"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "Anderson_42"       "Anderson_OD_51"    "Berry_393"        
##  [4] "Berry_OD_86"       "Blankley_380"      "Blankley_5"       
##  [7] "Bloom_OD_144"      "Bloom_RES_268"     "Bloom_RES_558"    
## [10] "Chen_HIV_4"        "Darboe_RISK_11"    "Dawany_HIV_251"   
## [13] "Duffy_23"          "Esmail_203"        "Esmail_82"        
## [16] "Esmail_OD_893"     "Estevez_133"       "Estevez_259"      
## [19] "Gjoen_10"          "Gjoen_7"           "Gliddon_2_OD_4"   
## [22] "Gliddon_HIV_3"     "Gliddon_OD_3"      "Gliddon_OD_4"     
## [25] "Gong_OD_4"         "Heycken_FAIL_22"   "Hoang_OD_13"      
## [28] "Hoang_OD_20"       "Hoang_OD_3"        "Huang_OD_13"      
## [31] "Jacobsen_3"        "Jenum_8"           "Kaforou_27"       
## [34] "Kaforou_OD_44"     "Kaforou_OD_53"     "Kulkarni_HIV_2"   
## [37] "LauxdaCosta_OD_3"  "Lee_4"             "Leong_24"         
## [40] "Leong_RISK_29"     "Maertzdorf_15"     "Maertzdorf_4"     
## [43] "Maertzdorf_OD_100" "Mendelsoh_RISK_11" "PennNich_RISK_6"  
## [46] "Qian_OD_17"        "Rajan_HIV_5"       "Roe_3"            
## [49] "Roe_OD_4"          "Sambarey_HIV_10"   "Singhania_OD_20"  
## [52] "Sivakumaran_11"    "Sloot_HIV_2"       "Suliman_4"        
## [55] "Suliman_RISK_2"    "Suliman_RISK_4"    "Sweeney_OD_3"     
## [58] "Thompson_9"        "Thompson_FAIL_13"  "Thompson_RES_5"   
## [61] "Tornheim_71"       "Tornheim_RES_25"   "Verhagen_10"      
## [64] "Walter_51"         "Walter_PNA_119"    "Walter_PNA_47"    
## [67] "Zak_RISK_16"       "Zhao_NANO_6"       "Zimmer_RES_3"</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## We can use all of these signatures for further analysis</span>
<span class="va">siglist_hivtb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Run the TBSignatureProfiler to score the signatures in the data</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/capture.output.html">capture.output</a></span><span class="op">(</span><span class="va">ssgsea_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/runTBsigProfiler.html">runTBsigProfiler</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">hivtb_data</span>,
                                                 useAssay <span class="op">=</span> <span class="st">"log_cpm"</span>,
                                                 signatures <span class="op">=</span> <span class="va">TBsignatures</span>,
                                                 algorithm <span class="op">=</span> <span class="st">"ssGSEA"</span>,
                                                 combineSigAndAlgorithm <span class="op">=</span> <span class="cn">TRUE</span>,
                                                 parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Running ssGSEA</code></pre>
<pre><code>## Warning in .filterFeatures(expr, method): 2204 genes with constant expression
## values throuhgout the samples.</code></pre>
<p>When a <code>SummarizedExperiment</code> is the format of the input data for <code>runTBsigprofiler</code>, the returned object is also of the <code>SummarizedExperiment</code>. The scores will be returned as a part of the colData.</p>
<p>Below, we subset the data to compare the enrichment scores for the Anderson_42, Anderson_OD_51, and Berry_393 signatures.</p>
<p>Signature Scores</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## New colData entries from the Profiler</span>
<span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Anderson_42"</span>, <span class="st">"Anderson_OD_51"</span>, <span class="st">"Berry_393"</span><span class="op">)</span>
<span class="va">ssgsea_print_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">ssgsea_result</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Disease"</span>, <span class="va">sigs</span><span class="op">)</span><span class="op">]</span>
<span class="va">ssgsea_print_results</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">ssgsea_print_results</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, <span class="fl">4</span><span class="op">)</span>

<span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">ssgsea_print_results</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-4589bb262dfc539b78d0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4589bb262dfc539b78d0">{"x":{"filter":"none","data":[["R01_1","R01_10","R01_11","R01_12","R01_13","R01_14","R01_15","R01_16","R01_2","R01_3","R01_4","R01_5","R01_6","R01_7","R01_8","R01_9","R02_17","R02_18","R02_19","R02_20","R02_21","R02_22","R02_23","R02_24","R02_25","R02_26","R02_27","R02_28","R02_29","R02_30","R02_31"],["tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","hiv_only","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only"],[0.225,0.2051,0.1842,0.2055,0.1925,0.199,0.2263,0.2206,0.2182,0.1912,0.1755,0.2032,0.207,0.2017,0.1917,0.2116,0.2117,0.2236,0.2235,0.2144,0.2403,0.1972,0.221,0.2216,0.2073,0.2269,0.2029,0.2042,0.213,0.206,0.1879],[0.2302,0.2837,0.2483,0.2496,0.2642,0.2808,0.2883,0.2372,0.2647,0.2798,0.2667,0.2441,0.248,0.2403,0.253,0.262,0.2484,0.2044,0.2343,0.2394,0.2404,0.2599,0.2482,0.247,0.2439,0.2394,0.2297,0.2608,0.2727,0.2522,0.2446],[0.3781,0.4015,0.3846,0.3996,0.3889,0.3984,0.3994,0.3437,0.3813,0.3944,0.3861,0.3805,0.3857,0.3916,0.3966,0.3979,0.369,0.3503,0.3618,0.3664,0.3448,0.3866,0.351,0.3722,0.3523,0.357,0.3639,0.3952,0.3935,0.3874,0.3719]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Disease<\/th>\n      <th>Anderson_42<\/th>\n      <th>Anderson_OD_51<\/th>\n      <th>Berry_393<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization-with-tbsignatureprofiler-plots" class="section level2 tabset">
<h2 class="hasAnchor">
<a href="#visualization-with-tbsignatureprofiler-plots" class="anchor"></a>Visualization with TBSignatureProfiler Plots</h2>
<div id="heatmap-with-all-signatures" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap-with-all-signatures" class="anchor"></a>Heatmap with all Signatures</h3>
<p>Commonly, enrichment scores are compared across signatures by means of a heatmap combined with clustering methods to group samples and/or scores. The <code>signatureHeatmap</code> function uses the information from the score data to visualize changes in gene expression across samples and signatures (or genes, if only one signature is selected).</p>
<p>Here, the columns of the heatmap represent samples, and rows represent signatures. Rows are split according to annotation data with associated signature disease type. As we move across the columns, we see different patterns of gene expression as indicated by the varying color and intensity of individual rectangles. In the top bar, the solid red represents a sample is HIV infected only, and solid blue indicates that the sample is both HIV and TB infected. In the gradient area of the heatmap, the scaled scores are associated with either up-regulated or down-regulated genes. A cluster of samples near the bottom of the heatmap reveals that some signatures are inversely associated with TB/HIV identification, as their phenotypic mapping for lower/higher scores is nearly the opposite of most of the other signatures.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Colors for gradient</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">6</span>, <span class="st">"Spectral"</span><span class="op">)</span>
<span class="va">col.me</span> <span class="op">&lt;-</span> <span class="fu">circlize</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html">colorRamp2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span>from <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, to <span class="op">=</span> <span class="fl">2</span>,
                                   length.out <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>, <span class="va">colors</span><span class="op">)</span>

<span class="fu"><a href="../../reference/signatureHeatmap.html">signatureHeatmap</a></span><span class="op">(</span><span class="va">ssgsea_result</span>, name <span class="op">=</span> <span class="st">"Heatmap of Signatures,
                 ssGSEA Algorithm"</span>,
                 signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,
                 annotationColNames <span class="op">=</span> <span class="st">"Disease"</span>,
                 scale <span class="op">=</span> <span class="cn">TRUE</span>,
                 showColumnNames <span class="op">=</span> <span class="cn">TRUE</span>,
                 choose_color <span class="op">=</span> <span class="va">col.me</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/all%20sigs%20heatmap-1.png" width="700"></p>
</div>
<div id="boxplots-of-scores-all-signatures" class="section level3">
<h3 class="hasAnchor">
<a href="#boxplots-of-scores-all-signatures" class="anchor"></a>Boxplots of Scores, All Signatures</h3>
<p>Another method of visualization for scores is that of boxplots. When multiple signatures in the input data are to be compared, the <code>signatureBoxplot</code> function takes the scores for each signature and produces an individual boxplot, with jittered points representing individual sample scores. For this specific example, it is clear that some signatures do a better job at differentiating the TB/HIV and HIV only samples than others, as seen by overlapping or separate spreads of the adjacent boxplots.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/signatureBoxplot.html">signatureBoxplot</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">ssgsea_result</span>,
                 name <span class="op">=</span> <span class="st">"Boxplots of Signatures, ssGSEA"</span>,
                 signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,
                 annotationColName <span class="op">=</span> <span class="st">"Disease"</span>, rotateLabels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/Boxplots%20all-1.png" width="1440"></p>
</div>
<div id="boxplots-of-scores-individual-signatures" class="section level3 tabset">
<h3 class="hasAnchor">
<a href="#boxplots-of-scores-individual-signatures" class="anchor"></a>Boxplots of Scores, Individual Signatures</h3>
<p>If only one boxplot of a signature is desired, then that signature alone should be specified in <code>signatureColNames</code>. Otherwise, an array of plots for all signatures will be produced. Here we use the signatures Anderson_42, Anderson_OD_51, and Berry_393.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">sigs</span><span class="op">)</span><span class="op">{</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"####"</span>, <span class="va">i</span>, <span class="st">"\n"</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="../../reference/signatureBoxplot.html">signatureBoxplot</a></span><span class="op">(</span><span class="va">ssgsea_result</span>, name <span class="op">=</span> <span class="va">i</span>, signatureColNames <span class="op">=</span> <span class="va">i</span>,
                 annotationColName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Disease"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n\n"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div id="anderson_42" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_42" class="anchor"></a>Anderson_42</h4>
<p><img src="TBSig_Vignette_files/figure-html/Boxplots%20indiv-1.png" width="700"></p>
</div>
<div id="anderson_od_51" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_od_51" class="anchor"></a>Anderson_OD_51</h4>
<p><img src="TBSig_Vignette_files/figure-html/Boxplots%20indiv-2.png" width="700"></p>
</div>
<div id="berry_393" class="section level4">
<h4 class="hasAnchor">
<a href="#berry_393" class="anchor"></a>Berry_393</h4>
<p><img src="TBSig_Vignette_files/figure-html/Boxplots%20indiv-3.png" width="700"></p>
</div>
</div>
<div id="heatmaps-for-individual-signatures" class="section level3 tabset">
<h3 class="hasAnchor">
<a href="#heatmaps-for-individual-signatures" class="anchor"></a>Heatmaps for Individual Signatures</h3>
<p>The <code>signatureGeneHeatmap</code> function produces heatmaps that highlight a single selected signature. The resulting plots feature samples on the columns, and the genes in the signature on the rows. The annotation rows at the top are related to both the disease phenotype and to the signature score itself. The gradient of the heatmap is created using the gene expression values in the assay specified by the <code>useAssay</code> parameter. Here, we observe how the log(CPM) values for specific genes relate to the signature score for each sample using the signatures Anderson_42, Anderson_OD_51, and Berry_393.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">sigs</span><span class="op">)</span><span class="op">{</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"####"</span>, <span class="va">i</span>, <span class="st">"\n"</span><span class="op">)</span>

  <span class="fu"><a href="../../reference/signatureGeneHeatmap.html">signatureGeneHeatmap</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">ssgsea_result</span>, useAssay <span class="op">=</span> <span class="st">"log_cpm"</span>,
                       <span class="va">TBsignatures</span><span class="op">[</span><span class="va">sigs</span><span class="op">]</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,
                       name <span class="op">=</span> <span class="va">i</span>, signatureColNames <span class="op">=</span> <span class="va">i</span>,
                       annotationColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Disease"</span><span class="op">)</span>,
                       showColumnNames <span class="op">=</span> <span class="cn">TRUE</span>,
                       choose_color <span class="op">=</span> <span class="va">col.me</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n\n"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div id="anderson_42-1" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_42-1" class="anchor"></a>Anderson_42</h4>
<p><img src="TBSig_Vignette_files/figure-html/Heatmaps%20indiv-1.png" width="700"></p>
</div>
<div id="anderson_od_51-1" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_od_51-1" class="anchor"></a>Anderson_OD_51</h4>
<p><img src="TBSig_Vignette_files/figure-html/Heatmaps%20indiv-2.png" width="700"></p>
</div>
<div id="berry_393-1" class="section level4">
<h4 class="hasAnchor">
<a href="#berry_393-1" class="anchor"></a>Berry_393</h4>
<p><img src="TBSig_Vignette_files/figure-html/Heatmaps%20indiv-3.png" width="700"></p>
</div>
</div>
<div id="compare-scoring-methods-for-a-single-signature" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-scoring-methods-for-a-single-signature" class="anchor"></a>Compare scoring methods for a single signature</h3>
<p>The <code>compareAlgs</code> function allows multiple scoring methods to be compared via a heatmap or boxplot with samples on the columns and methods on the rows. Here, we compare scoring methods for the “Anderson_42” signature. It seems that singscore may be the best method here, as its sample scores most closely align with the information provided by the annotation data.</p>
<p>An examination of the boxplot and heatmap together determine that PLAGE and the comparing Z-score methods are least helpful in correctly identifying TB from LTBI subjects - AUC scores are little more than 0.5 and subjects in different groups are falsely assigned similar scores. According to the boxplot, we see that ssGSEA has the highest predictive AUC.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Heatmap</span>
<span class="fu"><a href="../../reference/compareAlgs.html">compareAlgs</a></span><span class="op">(</span><span class="va">hivtb_data</span>, annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,
            scale <span class="op">=</span> <span class="cn">TRUE</span>,
            algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span>,
                          <span class="st">"singscore"</span>, <span class="st">"PLAGE"</span>, <span class="st">"Zscore"</span><span class="op">)</span>,
            useAssay <span class="op">=</span> <span class="st">"log_counts"</span>,
            signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Anderson_42"</span><span class="op">]</span>,
            choose_color <span class="op">=</span> <span class="va">col.me</span>, show.pb <span class="op">=</span> <span class="cn">FALSE</span>,
            parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/compareAlgs%20ex-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Boxplot</span>
<span class="fu"><a href="../../reference/compareAlgs.html">compareAlgs</a></span><span class="op">(</span><span class="va">hivtb_data</span>, annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,
            scale <span class="op">=</span> <span class="cn">TRUE</span>,
            algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span>,
                          <span class="st">"singscore"</span>, <span class="st">"PLAGE"</span>, <span class="st">"Zscore"</span><span class="op">)</span>,
            useAssay <span class="op">=</span> <span class="st">"log_counts"</span>,
            signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Anderson_42"</span><span class="op">]</span>,
            choose_color <span class="op">=</span> <span class="va">col.me</span>, show.pb <span class="op">=</span> <span class="cn">FALSE</span>,
            parallel.sz <span class="op">=</span> <span class="fl">1</span>, output <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/compareAlgs%20ex-2.png" width="700"></p>
</div>
</div>
<div id="comparing-scores-via-bootstrapped-aucs" class="section level2 tabset">
<h2 class="hasAnchor">
<a href="#comparing-scores-via-bootstrapped-aucs" class="anchor"></a>Comparing scores via bootstrapped AUCs</h2>
<div id="table-with-t-tests-auc" class="section level3">
<h3 class="hasAnchor">
<a href="#table-with-t-tests-auc" class="anchor"></a>Table with T-tests &amp; AUC</h3>
<p>The <code>TableAUC</code> function creates a table of useful information regarding the scored signatures. First, a 2-sample t-test is conducted on the scores with the TB/HIV status as the response variable. This is intended to give one metric of whether the signature is a valuable identifier of the disease phenotype. The log-scaled p-values are provided for easier means of comparison.</p>
<p>The table also gives the AUC values for the signature scores, along with confidence intervals derived from bootstrapping. Most AUC estimates are fairly high.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/tableAUC.html">tableAUC</a></span><span class="op">(</span><span class="va">ssgsea_result</span>,
         annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,
         signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,
         num.boot <span class="op">=</span> <span class="fl">100</span>,
         pb.show <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead</code></pre>
<div id="htmlwidget-2fbc71ce743880157adb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2fbc71ce743880157adb">{"x":{"filter":"none","data":[["Kulkarni_HIV_2","Thompson_9","Hoang_OD_3","Blankley_5","Heycken_FAIL_22","Hoang_OD_13","Blankley_380","Bloom_OD_144","Bloom_RES_268","Gong_OD_4","Mendelsoh_RISK_11","Kaforou_27","Sweeney_OD_3","Zimmer_RES_3","Darboe_RISK_11","Rajan_HIV_5","Walter_51","Zak_RISK_16","Hoang_OD_20","Tornheim_RES_25","Bloom_RES_558","Gliddon_2_OD_4","Gliddon_OD_4","Maertzdorf_15","Roe_3","Berry_393","Esmail_OD_893","Suliman_4","Sambarey_HIV_10","Estevez_259","Tornheim_71","Kaforou_OD_53","Estevez_133","LauxdaCosta_OD_3","Gliddon_HIV_3","Gliddon_OD_3","Esmail_203","Roe_OD_4","Maertzdorf_OD_100","Thompson_FAIL_13","Qian_OD_17","Suliman_RISK_2","Walter_PNA_47","Gjoen_7","Kaforou_OD_44","Anderson_OD_51","Jacobsen_3","Duffy_23","Huang_OD_13","Walter_PNA_119","Esmail_82","Leong_RISK_29","Anderson_42","Thompson_RES_5","Sloot_HIV_2","Verhagen_10","Leong_24","Chen_HIV_4","Dawany_HIV_251","Maertzdorf_4","PennNich_RISK_6","Berry_OD_86","Singhania_OD_20","Gjoen_10","Zhao_NANO_6","Suliman_RISK_4","Jenum_8","Sivakumaran_11","Lee_4"],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.0001,0,0.0003,0.0003,0,0,0,0.0002,0.0001,0.0001,0.0002,0.0001,0.0012,0.0005,0.001,0.0001,0.0001,0.0029,0.0005,0.0017,0.0009,0.0061,0.0068,0.0028,0.0028,0.0106,0.0096,0.0072,0.0156,0.0118,0.0054,0.0207,0.0164,0.0278,0.0571,0.0796,0.0646,0.0586,0.3052,0.0625,0.136,0.1372,0.2392,0.1568,0.2231,0.5014,0.2538,0.6886,0.8756,0.5818],[207.1333,148.0662,178.9241,142.9362,142.831,122.5744,120.0574,117.3455,128.8537,104.3571,125.3931,119.7161,101.7419,101.7419,115.2059,115.8811,112.2158,114.8476,112.9667,97.1836,109.219,81.48,81.48,125.3667,101.7238,102.3001,87.9775,90.7221,91.1867,87.9072,90.3759,67.051,76.58,69.4077,92.9589,92.9589,58.5309,75.6901,63.5168,70.177,50.922,49.8912,58.9577,58.9179,45.4278,46.4517,49.288,41.6131,44.4319,52.2314,38.7825,41.1099,35.8293,28.6366,25.3075,27.3968,28.3698,11.8671,27.7295,19.9503,19.8657,14.3044,18.5306,15.0028,6.9026,13.7115,3.7304,1.3282,5.416],[0.9719,0.8828,0.8665,0.8552,0.851,0.8266,0.8263,0.8293,0.8293,0.8029,0.8085,0.8065,0.8003,0.8003,0.7904,0.7667,0.794,0.7779,0.7815,0.7879,0.7569,0.7531,0.7531,0.7536,0.759,0.7559,0.738,0.7428,0.7317,0.7313,0.7263,0.6858,0.6904,0.6809,0.672,0.672,0.6605,0.6624,0.6628,0.6558,0.6171,0.6195,0.6255,0.6182,0.614,0.6062,0.6042,0.5766,0.5534,0.5677,0.5615,0.5406,0.5374,0.5153,0.5205,0.5012,0.483,0.4704,0.4713,0.4648,0.446,0.4479,0.4396,0.3983,0.3889,0.3849,0.3413,0.2871,0.2911],[0.9917,0.95,0.9458,0.9375,0.9333,0.925,0.9208,0.9208,0.9208,0.9208,0.9208,0.9083,0.9083,0.9083,0.9042,0.9,0.9,0.9,0.8958,0.8958,0.8875,0.8875,0.8875,0.8875,0.8875,0.8792,0.8708,0.8708,0.8667,0.8625,0.8583,0.8417,0.8375,0.8333,0.825,0.825,0.8208,0.8167,0.8125,0.8083,0.7917,0.7875,0.7875,0.7833,0.7833,0.775,0.775,0.7542,0.75,0.75,0.7458,0.7292,0.725,0.7083,0.7083,0.6958,0.6833,0.675,0.6708,0.6625,0.6542,0.65,0.6417,0.6042,0.6,0.5958,0.5542,0.5083,0.5042],[0.9917,0.95,0.9458,0.9375,0.9333,0.925,0.9208,0.9208,0.9208,0.9208,0.9208,0.9083,0.9083,0.9083,0.9042,0.9,0.9,0.9,0.8958,0.8958,0.8875,0.8875,0.8875,0.8875,0.8875,0.8792,0.8708,0.8708,0.8667,0.8625,0.8583,0.8417,0.8375,0.8333,0.825,0.825,0.8208,0.8167,0.8125,0.8083,0.7917,0.7875,0.7875,0.7833,0.7833,0.775,0.775,0.7542,0.75,0.75,0.7458,0.7292,0.725,0.7083,0.7083,0.6958,0.6833,0.675,0.6708,0.6625,0.6542,0.65,0.6417,0.6042,0.6,0.5958,0.5542,0.5083,0.5042]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Signature<\/th>\n      <th>P.value<\/th>\n      <th>neg10xLog.P.value.<\/th>\n      <th>LowerAUC<\/th>\n      <th>AUC<\/th>\n      <th>UpperAUC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="boxplots-for-the-bootstrapped-aucs" class="section level3">
<h3 class="hasAnchor">
<a href="#boxplots-for-the-bootstrapped-aucs" class="anchor"></a>Boxplots for the bootstrapped AUCs</h3>
<p>We can visualize the bootstrapped AUCs for the scored signatures by creating a collection of boxplots.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/compareBoxplots.html">compareBoxplots</a></span><span class="op">(</span><span class="va">ssgsea_result</span>, annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,
                signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,
                pb.show <span class="op">=</span> <span class="cn">FALSE</span>, rotateLabels <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead

## Warning in roc.default(predictor = score, response = annotationData): 'response'
## has more than two levels. Consider setting 'levels' explicitly or using
## 'multiclass.roc' instead</code></pre>
<p><img src="TBSig_Vignette_files/figure-html/BS%20Boxplots-1.png" width="700"></p>
</div>
<div id="roc-plots-for-all-signatures-with-95-ci-bands" class="section level3">
<h3 class="hasAnchor">
<a href="#roc-plots-for-all-signatures-with-95-ci-bands" class="anchor"></a>ROC plots for all signatures with 95% CI bands</h3>
<p>The <code>signatureROCplot_CI</code> function plots the ROC curves for signature scores in separate plots, along with CI bands for the specified level of confidence. The <code>signatureROCplot</code> function is similar, but does not include confidence interval bands.</p>
<p>Note that, in some cases, signatures will be positive identifiers of TB whereas others are negative identifiers, hence some ROC curves will be below the chance line.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/signatureROCplot_CI.html">signatureROCplot_CI</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">ssgsea_result</span>,
                   signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,
                   annotationColName <span class="op">=</span> <span class="st">"Disease"</span>, pb.show <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/ROC%20all-1.png" width="1152"></p>
</div>
<div id="separate-roc-plots-95-ci-bands" class="section level3 tabset">
<h3 class="hasAnchor">
<a href="#separate-roc-plots-95-ci-bands" class="anchor"></a>Separate ROC plots, 95% CI bands</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">sigs</span><span class="op">)</span><span class="op">{</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"####"</span>, <span class="va">i</span>, <span class="st">"\n"</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="../../reference/signatureROCplot_CI.html">signatureROCplot_CI</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">ssgsea_result</span>,
                   signatureColNames <span class="op">=</span> <span class="va">i</span>,
                   annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,
                   name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"ROC plot,"</span>, <span class="va">i</span>, sep <span class="op">=</span> <span class="st">" "</span><span class="op">)</span>, pb.show <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n\n"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div id="anderson_42-2" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_42-2" class="anchor"></a>Anderson_42</h4>
<p><img src="TBSig_Vignette_files/figure-html/ROC%20indiv-1.png" width="700"></p>
</div>
<div id="anderson_od_51-2" class="section level4">
<h4 class="hasAnchor">
<a href="#anderson_od_51-2" class="anchor"></a>Anderson_OD_51</h4>
<p><img src="TBSig_Vignette_files/figure-html/ROC%20indiv-2.png" width="700"></p>
</div>
<div id="berry_393-2" class="section level4">
<h4 class="hasAnchor">
<a href="#berry_393-2" class="anchor"></a>Berry_393</h4>
<p><img src="TBSig_Vignette_files/figure-html/ROC%20indiv-3.png" width="700"></p>
</div>
</div>
</div>
<div id="signature-evaluation-with-logistic-regression" class="section level2 tabset">
<h2 class="hasAnchor">
<a href="#signature-evaluation-with-logistic-regression" class="anchor"></a>Signature Evaluation with Logistic Regression</h2>
<p>Another method of signature evaluation featured in the TBSignatureProfiler package is available with the <code>SignatureQuantitative</code> function.</p>
<p>During the training process, we bootstrap the samples, cross-validate the logistic regression prediction of the TB HIV status of each sample, and calculate the validation set AUCs from each cross-validation iteration. In theory, if a signature is a useful predictor of status, it should have a higher average AUC from these logistic regression bootstraps.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Make an additional binary indicator for signature evaluation</span>
<span class="va">hivtb_data</span><span class="op">$</span><span class="va">BinaryTB</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">$</span><span class="va">Disease</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span>

<span class="co">## We can normalize the data and evaluate the signatures</span>
<span class="va">hivtb_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/deseq2_norm_rle.html">deseq2_norm_rle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">hivtb_data</span>, <span class="st">"counts"</span><span class="op">)</span><span class="op">)</span>

<span class="va">quantitative.result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/SignatureQuantitative.html">SignatureQuantitative</a></span><span class="op">(</span><span class="va">hivtb_norm</span>,
                       targetVec.num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span><span class="op">$</span><span class="va">BinaryTB</span>,
                       signature.list <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="va">sigs</span><span class="op">]</span>,
                       signature.name.vec <span class="op">=</span> <span class="va">sigs</span>,
                       num.boot <span class="op">=</span> <span class="fl">3</span>,
                       pb.show <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>We also create a boxplot of AUC for each signature, which provides a clear comparison between the signatures.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/plotQuantitative.html">plotQuantitative</a></span><span class="op">(</span><span class="va">hivtb_norm</span>, targetVec.num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span><span class="op">$</span><span class="va">BinaryTB</span>,
                       signature.list <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="va">sigs</span><span class="op">]</span>,
                       signature.name.vec <span class="op">=</span> <span class="va">sigs</span>,
                       num.boot <span class="op">=</span> <span class="fl">3</span>, pb.show <span class="op">=</span> <span class="cn">FALSE</span>, rotateLabels <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="TBSig_Vignette_files/figure-html/sigeval%20plot-1.png" width="700"></p>
<div id="table-of-auc-results" class="section level3">
<h3 class="hasAnchor">
<a href="#table-of-auc-results" class="anchor"></a>Table of AUC Results</h3>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">quantitative.result</span><span class="op">$</span><span class="va">df.auc.ci</span>, <span class="fl">4</span><span class="op">)</span>,
              options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span>, pageLength <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
              rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-125117cf6992a0487236" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-125117cf6992a0487236">{"x":{"filter":"none","data":[[0.7625,0.8458,0.8458],[0.8984,0.9467,0.9126],[0.997,0.9948,0.992],[-0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Estimate<\/th>\n      <th>CI lower.5%<\/th>\n      <th>CI upper.95%<\/th>\n      <th>Std. Error<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2,3]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-of-sensitivity-results" class="section level3">
<h3 class="hasAnchor">
<a href="#table-of-sensitivity-results" class="anchor"></a>Table of Sensitivity Results</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">quantitative.result</span><span class="op">$</span><span class="va">df.sensitivity.ci</span>, <span class="fl">4</span><span class="op">)</span>,
              options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span>, pageLength <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
              rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-1017778dc99b0c349eb6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1017778dc99b0c349eb6">{"x":{"filter":"none","data":[[0.625,0.875,0.875],[0.6944,0.8675,0.8643],[0.9494,0.8975,0.9929],[0,0,-0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Estimate.Sensitivity<\/th>\n      <th>CI lower.5%<\/th>\n      <th>CI upper.95%<\/th>\n      <th>Std. Error<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2,3]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-of-specificity-results" class="section level3">
<h3 class="hasAnchor">
<a href="#table-of-specificity-results" class="anchor"></a>Table of Specificity Results</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">quantitative.result</span><span class="op">$</span><span class="va">df.specificity.ci</span>, <span class="fl">4</span><span class="op">)</span>,
              options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span>, pageLength <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
              rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-3f33bfc4a48d1ec785dc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3f33bfc4a48d1ec785dc">{"x":{"filter":"none","data":[[0.8,0.8,0.7333],[0.8615,0.9571,0.7816],[1,1,0.9404],[0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Estimate.Specificity<\/th>\n      <th>CI lower.5%<\/th>\n      <th>CI upper.95%<\/th>\n      <th>Std. Error<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2,3]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19042)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] TBSignatureProfiler_1.5.0   SummarizedExperiment_1.22.0
##  [3] Biobase_2.52.0              GenomicRanges_1.44.0       
##  [5] GenomeInfoDb_1.28.0         IRanges_2.26.0             
##  [7] S4Vectors_0.30.0            BiocGenerics_0.38.0        
##  [9] MatrixGenerics_1.4.0        matrixStats_0.59.0         
## [11] BiocStyle_2.20.2           
## 
## loaded via a namespace (and not attached):
##   [1] circlize_0.4.13             systemfonts_1.0.2          
##   [3] plyr_1.8.6                  splines_4.1.0              
##   [5] GSEABase_1.54.0             BiocParallel_1.26.0        
##   [7] crosstalk_1.1.1             ggplot2_3.3.4              
##   [9] digest_0.6.27               foreach_1.5.1              
##  [11] htmltools_0.5.1.1           gdata_2.18.0               
##  [13] fansi_0.5.0                 magrittr_2.0.1             
##  [15] memoise_2.0.0               ScaledMatrix_1.0.0         
##  [17] cluster_2.1.2               doParallel_1.0.16          
##  [19] limma_3.48.0                recipes_0.1.16             
##  [21] ComplexHeatmap_2.8.0        Biostrings_2.60.1          
##  [23] annotate_1.70.0             gower_0.2.2                
##  [25] pkgdown_1.6.1               colorspace_2.0-1           
##  [27] blob_1.2.1                  textshaping_0.3.5          
##  [29] xfun_0.24                   dplyr_1.0.7                
##  [31] crayon_1.4.1                RCurl_1.98-1.3             
##  [33] jsonlite_1.7.2              graph_1.70.0               
##  [35] genefilter_1.74.0           survival_3.2-11            
##  [37] iterators_1.0.13            glue_1.4.2                 
##  [39] gtable_0.3.0                ipred_0.9-11               
##  [41] zlibbioc_1.38.0             XVector_0.32.0             
##  [43] GetoptLong_1.0.5            DelayedArray_0.18.0        
##  [45] BiocSingular_1.8.1          Rhdf5lib_1.14.1            
##  [47] shape_1.4.6                 SingleCellExperiment_1.14.1
##  [49] HDF5Array_1.20.0            scales_1.1.1               
##  [51] DBI_1.1.1                   edgeR_3.34.0               
##  [53] Rcpp_1.0.6                  xtable_1.8-4               
##  [55] clue_0.3-59                 proxy_0.4-26               
##  [57] bit_4.0.4                   rsvd_1.0.5                 
##  [59] lava_1.6.9                  prodlim_2019.11.13         
##  [61] GSVA_1.40.1                 DT_0.18                    
##  [63] glmnet_4.1-1                htmlwidgets_1.5.3          
##  [65] httr_1.4.2                  RColorBrewer_1.1-2         
##  [67] ellipsis_0.3.2              pkgconfig_2.0.3            
##  [69] XML_3.99-0.6                farver_2.1.0               
##  [71] nnet_7.3-16                 sass_0.4.0                 
##  [73] caret_6.0-88                locfit_1.5-9.4             
##  [75] utf8_1.2.1                  tidyselect_1.1.1           
##  [77] labeling_0.4.2              rlang_0.4.11               
##  [79] reshape2_1.4.4              AnnotationDbi_1.54.1       
##  [81] munsell_0.5.0               tools_4.1.0                
##  [83] cachem_1.0.5                generics_0.1.0             
##  [85] RSQLite_2.2.7               evaluate_0.14              
##  [87] stringr_1.4.0               fastmap_1.1.0              
##  [89] yaml_2.2.1                  ragg_1.1.3                 
##  [91] ModelMetrics_1.2.2.2        knitr_1.33                 
##  [93] bit64_4.0.5                 fs_1.5.0                   
##  [95] purrr_0.3.4                 KEGGREST_1.32.0            
##  [97] nlme_3.1-152                sparseMatrixStats_1.4.0    
##  [99] compiler_4.1.0              png_0.1-7                  
## [101] e1071_1.7-7                 geneplotter_1.70.0         
## [103] tibble_3.1.2                bslib_0.2.5.1              
## [105] stringi_1.6.2               highr_0.9                  
## [107] desc_1.3.0                  lattice_0.20-44            
## [109] Matrix_1.3-4                vctrs_0.3.8                
## [111] pillar_1.6.1                lifecycle_1.0.0            
## [113] rhdf5filters_1.4.0          BiocManager_1.30.16        
## [115] ROCit_2.1.1                 jquerylib_0.1.4            
## [117] GlobalOptions_0.1.2         data.table_1.14.0          
## [119] bitops_1.0-7                irlba_2.3.3                
## [121] R6_2.5.0                    bookdown_0.22              
## [123] codetools_0.2-18            MASS_7.3-54                
## [125] gtools_3.9.2                assertthat_0.2.1           
## [127] rhdf5_2.36.0                DESeq2_1.32.0              
## [129] rprojroot_2.0.2             rjson_0.2.20               
## [131] withr_2.4.2                 GenomeInfoDbData_1.2.6     
## [133] rpart_4.1-15                timeDate_3043.102          
## [135] grid_4.1.0                  beachmat_2.8.0             
## [137] class_7.3-19                tidyr_1.1.3                
## [139] rmarkdown_2.9               DelayedMatrixStats_1.14.0  
## [141] Cairo_1.5-12.2              singscore_1.12.0           
## [143] pROC_1.17.0.1               lubridate_1.7.10</code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Jenkins, Aubrey Odom, Xutao Wang, Yue Zhao, W. Evan Johnson, Christian Love.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
