<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TBSignatureProfiler">
<title>Introduction to the TBSignatureProfiler • TBSignatureProfiler</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the TBSignatureProfiler">
<meta property="og:description" content="TBSignatureProfiler">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TBSignatureProfiler</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.17.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/rmd/TBSig_Vignette.html">Command Line Analysis</a>
    <a class="dropdown-item" href="../articles/rmd/OriginalModelTutorial.html">Original Models</a>
    <a class="dropdown-item" href="../articles/rmd/shinytutorial.html">Interactive Shiny Analysis</a>
    <a class="dropdown-item" href="../articles/rmd/SignatureAddition.html">Adding Signatures to the Profiler</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">TBSP News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/TBSignatureProfiler">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<link href="tbspVignette_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="tbspVignette_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="tbspVignette_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="tbspVignette_files/datatables-binding-0.33/datatables.js"></script><link href="tbspVignette_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="tbspVignette_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="tbspVignette_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="tbspVignette_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="tbspVignette_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to the TBSignatureProfiler</h1>
                        <h4 data-toc-skip class="author">Aubrey
Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">W. Evan
Johnson</h4>
            <address class="author_afil">
      Division of Infectious Disease, Center for Data Science, Rutgers
University – New Jersey Medical School, Newark, NJ,
USA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wj183@njms.rutgers.edu" class="email">wj183@njms.rutgers.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">May 7, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wejlab/TBSignatureProfiler/blob/HEAD/vignettes/tbspVignette.Rmd" class="external-link"><code>vignettes/tbspVignette.Rmd</code></a></small>
      <div class="d-none name"><code>tbspVignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction-to-the-tbsignatureprofiler">Introduction to the TBSignatureProfiler<a class="anchor" aria-label="anchor" href="#introduction-to-the-tbsignatureprofiler"></a>
</h2>
<p>Tuberculosis (TB) is the leading cause of infectious disease
mortality worldwide, causing on average nearly 1.4 million deaths per
year. A consistent issue faced in controlling TB outbreak is difficulty
in diagnosing individuals with particular types of TB infections for
which bacteria tests (e.g., via GeneXpert, sputum) prove inaccurate. As
an alternative mechanism of diagnosis for these infections, researchers
have discovered and published multiple gene expression signatures as
blood-based disease biomarkers. In this context, gene signatures are
defined as a combined group of genes with a uniquely characteristic
pattern of gene expression that occurs as a result of a medical
condition. To date, more than 75 signatures have been published by
researchers, though most have relatively low cross-condition validation
(e.g., testing TB in samples from diverse geographic and comorbidity
backgrounds). Furthermore, these signatures have never been formally
collected and made available as a single unified resource.</p>
<p>We aim to provide the scientific community with a resource to access
these aggregated signatures and to create an efficient means for their
visual and quantitative comparison via open source software. This
necessitated the development of the TBSignatureProfiler, a novel R
package which delivers a computational profiling platform for
researchers to characterize the diagnostic ability of existing
signatures in multiple comorbidity settings. This software allows for
signature strength estimation via several enrichment methods and
subsequent visualization of single- and multi-pathway results. Its
signature evaluation functionalities include signature profiling, AUC
bootstrapping, and leave-one-out cross-validation (LOOCV) of logistic
regression to approximate TB samples’ status. Its plotting
functionalities include sample-signature score heatmaps, bootstrap AUC
and LOOCV boxplots, and tables for presenting results.</p>
<p>More recently, the TBSignatureProfiler has undertaken a new role in
analyzing signatures across multiple chronic airway diseases, the most
recent being COVID-19 (see the <code>COVIDsignatures</code> object). As
we grow and expand the TBSignatureProfiler, we hope to add signatures
from multiple diseases to improve the package’s utility in the area of
gene signature comparison.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>In order to install the TBSignatureProfiler from Bioconductor, run
the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"TBSignatureProfiler"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="compatibility-with-summarizedexperiment-objects">Compatibility with SummarizedExperiment objects<a class="anchor" aria-label="anchor" href="#compatibility-with-summarizedexperiment-objects"></a>
</h2>
<p>While the TBSignatureProfiler often allows for the form of a
<code>data.frame</code> or <code>matrix</code> as input data, the most
ideal form of input for this package is that of the
<code>SummarizedExperiment</code> object. This is an amazing data
structure that is being developed by the as part of the <em><a href="https://bioconductor.org/packages/3.19/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
package, and is imported as part of the TBSignatureProfiler package. It
is able to store data matrices along with annotation information,
metadata, and reduced dimensionality data (PCA, t-SNE, etc.). To learn
more about proper usage and context of the
<code>SummarizedExperiment</code> object, you may want to take a look at
the <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">package
vignette</a>. A basic understanding of the <code>assay</code> and
<code>colData</code> properties of a <code>SummarizedExperiment</code>
will be useful for the purposes of this vignette.</p>
</div>
<div class="section level2">
<h2 id="a-quick-tutorial-for-the-tbsignatureprofiler">A Quick Tutorial for the TBSignatureProfiler<a class="anchor" aria-label="anchor" href="#a-quick-tutorial-for-the-tbsignatureprofiler"></a>
</h2>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wejlab/TBSignatureProfiler" class="external-link">TBSignatureProfiler</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-shiny-app">Run Shiny App<a class="anchor" aria-label="anchor" href="#run-shiny-app"></a>
</h3>
<p>This command is to start the TBSignatureProfiler shiny app. The Shiny
app implements several functions described in this vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/TBSPapp.html">TBSPapp</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The basic functions of the shiny app are also included in the command
line version of the TBSignatureProfiler, which is the focus of the
remainder of this vignette.</p>
</div>
<div class="section level3">
<h3 id="load-dataset-from-a-summarizedexperiment-object">Load dataset from a SummarizedExperiment object<a class="anchor" aria-label="anchor" href="#load-dataset-from-a-summarizedexperiment-object"></a>
</h3>
<p>In this tutorial, we will work with HIV and Tuberculosis (TB) gene
expression data in a <code>SummarizedExperiment</code> format. This
dataset is included in the TBSignatureProfiler package and can be loaded
into the global environment with <code>data("TB_hiv")</code>. The 31
samples in the dataset are marked as either having both TB and HIV
infection, or HIV infection only.</p>
<p>We begin by examining the dataset, which contains a matrix of counts
information (an “assay” in SummarizedExperiment terms) and another
matrix of meta data information on our samples (the “colData”). We will
also generate a few additional assays; these are the log(counts), the
counts per million (CPM) reads mapped, and the log(CPM) assays.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## HIV/TB gene expression data, included in the package</span></span>
<span><span class="va">hivtb_data</span> <span class="op">&lt;-</span> <span class="va">TB_hiv</span></span>
<span></span>
<span><span class="co">### Note that we have 25,369 genes, 33 samples, and 1 assay of counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 25369    31</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We start with only one assay</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of length 1</span></span>
<span><span class="co">## names(1): counts</span></span></code></pre>
<p>We now make a log counts, CPM and log CPM assay.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Make a log counts, CPM and log CPM assay</span></span>
<span><span class="va">hivtb_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mkAssay.html">mkAssay</a></span><span class="op">(</span><span class="va">hivtb_data</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, counts_to_CPM <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Check to see that we now have 4 assays</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">hivtb_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of length 4</span></span>
<span><span class="co">## names(4): counts log_counts counts_cpm log_counts_cpm</span></span></code></pre>
</div>
<div class="section level3">
<h3 class="tabset" id="profile-the-data">Profile the data<a class="anchor" aria-label="anchor" href="#profile-the-data"></a>
</h3>
<p>The TBSignatureProfiler enables comparison of multiple Tuberculosis
gene signatures. The package currently contains information on 79
signatures for comparison. The default signature list object for most
functions here is <code>TBsignatures</code>, although a list with
publication-given signature names is also available as
<code>TBcommon</code>. Data frames of annotation information for these
signatures, including information on associated disease and tissue type,
can be accessed as <code>sigAnnotData</code> and
<code>common_sigAnnotData</code> respectively.</p>
<p>With the <code>runTBSigProfiler</code> function, we are able to score
these signatures with a selection of algorithms, including <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7" class="external-link">gene
set variation analysis</a> (GSVA) (Hänzelmann et al, 2013), <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7" class="external-link">single-sample
GSEA</a> (ssGSEA) (Barbie et al, 2009), and the <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7" class="external-link">ASSIGN</a>
pathway profiling toolkit (Shen et al, 2015). For a complete list of
included scoring methods, run <code><a href="../reference/runTBsigProfiler.html">?runTBsigProfiler</a></code> in the
terminal.</p>
<p>Here, we evaluate all signatures included in the package with ssGSEA.
Paraphrasing from the <a href="http://software.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection/4" class="external-link">ssGSEA
documentation</a>, for each pairing of one of the 31 samples and its
gene set, ssGSEA calculates a separate enrichment score independent of
the phenotypic labeling (in this case, whether a sample has HIV/TB, or
HIV only). The single sample’s gene expression profile is then
transformed to a gene set enrichment profile. A score from the set
profile represents the activity level of the biological process in which
the gene set’s members are coordinately up- or down-regulated.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## List all signatures in the profiler</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"TBsignatures"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "Anderson_42"       "Anderson_OD_51"    "Berry_393"        </span></span>
<span><span class="co">##  [4] "Berry_OD_86"       "Blankley_380"      "Blankley_5"       </span></span>
<span><span class="co">##  [7] "Bloom_OD_144"      "Bloom_RES_268"     "Bloom_RES_558"    </span></span>
<span><span class="co">## [10] "Chen_5"            "Chen_HIV_4"        "Chendi_HIV_2"     </span></span>
<span><span class="co">## [13] "Darboe_RISK_11"    "Dawany_HIV_251"    "Duffy_23"         </span></span>
<span><span class="co">## [16] "Esmail_203"        "Esmail_82"         "Esmail_OD_893"    </span></span>
<span><span class="co">## [19] "Estevez_133"       "Estevez_259"       "Francisco_OD_2"   </span></span>
<span><span class="co">## [22] "Gjoen_10"          "Gjoen_7"           "Gliddon_2_OD_4"   </span></span>
<span><span class="co">## [25] "Gliddon_HIV_3"     "Gliddon_OD_3"      "Gliddon_OD_4"     </span></span>
<span><span class="co">## [28] "Gong_OD_4"         "Heycken_FAIL_22"   "Hoang_OD_13"      </span></span>
<span><span class="co">## [31] "Hoang_OD_20"       "Hoang_OD_3"        "Huang_OD_13"      </span></span>
<span><span class="co">## [34] "Jacobsen_3"        "Jenum_8"           "Kaforou_27"       </span></span>
<span><span class="co">## [37] "Kaforou_OD_44"     "Kaforou_OD_53"     "Kaul_3"           </span></span>
<span><span class="co">## [40] "Kulkarni_HIV_2"    "Kwan_186"          "LauxdaCosta_OD_3" </span></span>
<span><span class="co">## [43] "Lee_4"             "Leong_24"          "Leong_RISK_29"    </span></span>
<span><span class="co">## [46] "Long_RES_10"       "Maertzdorf_15"     "Maertzdorf_4"     </span></span>
<span><span class="co">## [49] "Maertzdorf_OD_100" "Natarajan_7"       "PennNich_RISK_6"  </span></span>
<span><span class="co">## [52] "Qian_OD_17"        "Rajan_HIV_5"       "Roe_3"            </span></span>
<span><span class="co">## [55] "Roe_OD_4"          "Sambarey_HIV_10"   "Singhania_OD_20"  </span></span>
<span><span class="co">## [58] "Sivakumaran_11"    "Sloot_HIV_2"       "Suliman_4"        </span></span>
<span><span class="co">## [61] "Suliman_RISK_2"    "Suliman_RISK_4"    "Sweeney_OD_3"     </span></span>
<span><span class="co">## [64] "Tabone_OD_11"      "Tabone_RES_25"     "Tabone_RES_27"    </span></span>
<span><span class="co">## [67] "Thompson_9"        "Thompson_FAIL_13"  "Thompson_RES_5"   </span></span>
<span><span class="co">## [70] "Tornheim_71"       "Tornheim_RES_25"   "Vargas_18"        </span></span>
<span><span class="co">## [73] "Vargas_42"         "Verhagen_10"       "Walter_51"        </span></span>
<span><span class="co">## [76] "Walter_PNA_119"    "Walter_PNA_47"     "Zak_RISK_16"      </span></span>
<span><span class="co">## [79] "Zhao_NANO_6"</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## We can use all of these signatures for further analysis</span></span>
<span><span class="va">siglist_hivtb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run the TBSignatureProfiler to score the signatures in the data</span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/capture.output.html" class="external-link">capture.output</a></span><span class="op">(</span><span class="va">ssgsea_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runTBsigProfiler.html">runTBsigProfiler</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">hivtb_data</span>,</span>
<span>                                                 useAssay <span class="op">=</span> <span class="st">"log_counts_cpm"</span>,</span>
<span>                                                 signatures <span class="op">=</span> <span class="va">TBsignatures</span>,</span>
<span>                                                 algorithm <span class="op">=</span> <span class="st">"ssGSEA"</span>,</span>
<span>                                                 combineSigAndAlgorithm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                                 parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Parameter update_genes is TRUE. Gene names will be updated.</span></span></code></pre>
<pre><code><span><span class="co">## The following signatures have &lt;2 genes that coincide with the genes in the given sample and will not be scored: Chendi_HIV_2</span></span></code></pre>
<pre><code><span><span class="co">## Running ssGSEA</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Remove any signatures that were not scored</span></span>
<span><span class="va">TBsignatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">TBsignatures</span>, <span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chendi_HIV_2"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>When a <code>SummarizedExperiment</code> is the format of the input
data for <code>runTBsigprofiler</code>, the returned object is also of
the <code>SummarizedExperiment</code>. The scores will be returned as a
part of the colData.</p>
<p>Below, we subset the data to compare the enrichment scores for the
Anderson_42, Anderson_OD_51, and Zak_RISK_16 signatures.</p>
<p>Signature Scores</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## New colData entries from the Profiler</span></span>
<span><span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Anderson_42"</span>, <span class="st">"Anderson_OD_51"</span>, <span class="st">"Zak_RISK_16"</span><span class="op">)</span></span>
<span><span class="va">ssgsea_print_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">ssgsea_result</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Disease"</span>, <span class="va">sigs</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">ssgsea_print_results</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">ssgsea_print_results</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">ssgsea_print_results</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"filter":"none","vertical":false,"data":[["R01_1","R01_10","R01_11","R01_12","R01_13","R01_14","R01_15","R01_16","R01_2","R01_3","R01_4","R01_5","R01_6","R01_7","R01_8","R01_9","R02_17","R02_18","R02_19","R02_20","R02_21","R02_22","R02_23","R02_24","R02_25","R02_26","R02_27","R02_28","R02_29","R02_30","R02_31"],["tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","hiv_only","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","tb_hiv","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only","hiv_only"],[0.2248,0.2049,0.1841,0.2053,0.1923,0.1988,0.2261,0.2204,0.218,0.1911,0.1753,0.2031,0.2069,0.2015,0.1916,0.2114,0.2115,0.2234,0.2233,0.2142,0.2401,0.197,0.2208,0.2214,0.2071,0.2267,0.2027,0.204,0.2128,0.2058,0.1877],[0.23,0.2834,0.2481,0.2493,0.264,0.2806,0.2881,0.237,0.2644,0.2796,0.2665,0.2439,0.2478,0.2401,0.2528,0.2618,0.2482,0.2042,0.2341,0.2392,0.2402,0.2596,0.248,0.2468,0.2437,0.2392,0.2295,0.2606,0.2724,0.252,0.2444],[0.4584,0.4939,0.4448,0.4863,0.4999,0.4967,0.4784,0.3774,0.4286,0.4777,0.4834,0.4686,0.4715,0.4985,0.4953,0.4939,0.4264,0.3766,0.3881,0.4115,0.3671,0.4267,0.3991,0.4321,0.4019,0.4201,0.4035,0.4985,0.4831,0.4613,0.425]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Disease<\/th>\n      <th>Anderson_42<\/th>\n      <th>Anderson_OD_51<\/th>\n      <th>Zak_RISK_16<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Disease","targets":1},{"name":"Anderson_42","targets":2},{"name":"Anderson_OD_51","targets":3},{"name":"Zak_RISK_16","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 class="tabset" id="visualization-with-tbsignatureprofiler-plots">Visualization with TBSignatureProfiler Plots<a class="anchor" aria-label="anchor" href="#visualization-with-tbsignatureprofiler-plots"></a>
</h3>
<div class="section level4">
<h4 id="heatmap-with-all-signatures">Heatmap with all Signatures<a class="anchor" aria-label="anchor" href="#heatmap-with-all-signatures"></a>
</h4>
<p>Commonly, enrichment scores are compared across signatures by means
of a heatmap combined with clustering methods to group samples and/or
scores. The <code>signatureHeatmap</code> function uses the information
from the score data to visualize changes in gene expression across
samples and signatures (or genes, if only one signature is
selected).</p>
<p>Here, the columns of the heatmap represent samples, and rows
represent signatures. Rows are split according to annotation data with
associated signature disease type. As we move across the columns, we see
different patterns of gene expression as indicated by the varying color
and intensity of individual rectangles. In the top bar, the solid red
represents a sample is HIV infected only, and solid blue indicates that
the sample is both HIV and TB infected. In the gradient area of the
heatmap, the scaled scores are associated with either up-regulated or
down-regulated genes.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Colors for gradient</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">6</span>, <span class="st">"Spectral"</span><span class="op">)</span></span>
<span><span class="va">col.me</span> <span class="op">&lt;-</span> <span class="fu">circlize</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html" class="external-link">colorRamp2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, to <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                   length.out <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>, <span class="va">colors</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/signatureHeatmap.html">signatureHeatmap</a></span><span class="op">(</span><span class="va">ssgsea_result</span>, name <span class="op">=</span> <span class="st">"Heatmap of Signatures,</span></span>
<span><span class="st">                 ssGSEA Algorithm"</span>,</span>
<span>                 signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,</span>
<span>                 annotationColNames <span class="op">=</span> <span class="st">"Disease"</span>,</span>
<span>                 scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 showColumnNames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 choose_color <span class="op">=</span> <span class="va">col.me</span><span class="op">)</span></span></code></pre></div>
<p><img src="tbspVignette_files/figure-html/all_sigs_heatmap-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="boxplots-of-scores-all-signatures">Boxplots of Scores, All Signatures<a class="anchor" aria-label="anchor" href="#boxplots-of-scores-all-signatures"></a>
</h4>
<p>Another method of visualization for scores is that of boxplots. When
multiple signatures in the input data are to be compared, the
<code>signatureBoxplot</code> function takes the scores for each
signature and produces an individual boxplot, with jittered points
representing individual sample scores. For this specific example, it is
clear that some signatures do a better job at differentiating the TB/HIV
and HIV only samples than others, as seen by overlapping or separate
spreads of the adjacent boxplots.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/signatureBoxplot.html">signatureBoxplot</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">ssgsea_result</span>,</span>
<span>                 name <span class="op">=</span> <span class="st">"Boxplots of Signatures, ssGSEA"</span>,</span>
<span>                 signatureColNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">)</span>,</span>
<span>                 annotationColName <span class="op">=</span> <span class="st">"Disease"</span>, rotateLabels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Use of `boxplotdfm$Group` is discouraged.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `Group` instead.</span></span>
<span><span class="co">## Use of `boxplotdfm$Group` is discouraged.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `Group` instead.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Use of `boxplotdfm$Score` is discouraged.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `Score` instead.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Use of `boxplotdfm$Group` is discouraged.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `Group` instead.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Use of `boxplotdfm$Score` is discouraged.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Use `Score` instead.</span></span></code></pre>
<p><img src="tbspVignette_files/figure-html/Boxplots_all-1.png" width="1440"></p>
</div>
<div class="section level4">
<h4 id="compare-scoring-methods-for-a-single-signature">Compare scoring methods for a single signature<a class="anchor" aria-label="anchor" href="#compare-scoring-methods-for-a-single-signature"></a>
</h4>
<p>The <code>compareAlgs</code> function allows multiple scoring methods
to be compared via a heatmap or boxplot with samples on the columns and
methods on the rows. Here, we compare scoring methods for the
“Tabone_RES_25” signature. It seems that singscore may be the best
method here, as its sample scores most closely align with the
information provided by the annotation data.</p>
<p>An examination of the boxplot and heatmap together determine that
PLAGE and the comparing Z-score methods are least helpful in correctly
identifying TB from LTBI subjects, although still quite good - average
AUC scores are a little less than 0.8 and subjects in different groups
are falsely assigned similar scores. According to the boxplot, we see
that singscore has the highest predictive AUC next to ssGSEA.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Heatmap</span></span>
<span><span class="fu"><a href="../reference/compareAlgs.html">compareAlgs</a></span><span class="op">(</span><span class="va">hivtb_data</span>, annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,</span>
<span>            scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span>,</span>
<span>                          <span class="st">"singscore"</span>, <span class="st">"PLAGE"</span>, <span class="st">"Zscore"</span><span class="op">)</span>,</span>
<span>            useAssay <span class="op">=</span> <span class="st">"log_counts"</span>,</span>
<span>            signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Tabone_RES_25"</span><span class="op">]</span>,</span>
<span>            choose_color <span class="op">=</span> <span class="va">col.me</span>, show.pb <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="tbspVignette_files/figure-html/compareAlgs_ex-1.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Boxplot</span></span>
<span><span class="fu"><a href="../reference/compareAlgs.html">compareAlgs</a></span><span class="op">(</span><span class="va">hivtb_data</span>, annotationColName <span class="op">=</span> <span class="st">"Disease"</span>,</span>
<span>            scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span>,</span>
<span>                          <span class="st">"singscore"</span>, <span class="st">"PLAGE"</span>, <span class="st">"Zscore"</span><span class="op">)</span>,</span>
<span>            useAssay <span class="op">=</span> <span class="st">"log_counts"</span>,</span>
<span>            signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Tabone_RES_25"</span><span class="op">]</span>,</span>
<span>            choose_color <span class="op">=</span> <span class="va">col.me</span>, show.pb <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            parallel.sz <span class="op">=</span> <span class="fl">1</span>, output <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span></span></code></pre></div>
<p><img src="tbspVignette_files/figure-html/compareAlgs_ex-2.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="additional-documentation-and-tutorials">Additional Documentation and Tutorials<a class="anchor" aria-label="anchor" href="#additional-documentation-and-tutorials"></a>
</h2>
<p>The functions presented as part of this vignette are but a limited
subset of those available in the TBSignatureProfiler package. A more
complete tutorial is available by going to <a href="https://wejlab.github.io/TBSignatureProfiler-docs/index.html" class="external-link">our
website</a>, and selecting the tab labeled “Command Line Analysis.”</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: AlmaLinux 8.9 (Midnight Oncilla)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: FlexiBLAS NETLIB;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">##  [3] GenomicRanges_1.56.0        GenomeInfoDb_1.40.0        </span></span>
<span><span class="co">##  [5] IRanges_2.38.0              S4Vectors_0.42.0           </span></span>
<span><span class="co">##  [7] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      </span></span>
<span><span class="co">##  [9] matrixStats_1.3.0           TBSignatureProfiler_1.17.1 </span></span>
<span><span class="co">## [11] BiocStyle_2.32.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3          rstudioapi_0.16.0          </span></span>
<span><span class="co">##   [3] jsonlite_1.8.8              shape_1.4.6.1              </span></span>
<span><span class="co">##   [5] magrittr_2.0.3              magick_2.8.3               </span></span>
<span><span class="co">##   [7] farver_2.1.1                rmarkdown_2.26             </span></span>
<span><span class="co">##   [9] GlobalOptions_0.1.2         fs_1.6.4                   </span></span>
<span><span class="co">##  [11] zlibbioc_1.50.0             ragg_1.3.0                 </span></span>
<span><span class="co">##  [13] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">##  [15] Cairo_1.6-2                 htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [17] S4Arrays_1.4.0              Rhdf5lib_1.26.0            </span></span>
<span><span class="co">##  [19] pROC_1.18.5                 SparseArray_1.4.0          </span></span>
<span><span class="co">##  [21] rhdf5_2.48.0                sass_0.4.9                 </span></span>
<span><span class="co">##  [23] bslib_0.7.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [25] desc_1.4.3                  HGNChelper_0.8.1           </span></span>
<span><span class="co">##  [27] plyr_1.8.9                  cachem_1.0.8               </span></span>
<span><span class="co">##  [29] lifecycle_1.0.4             iterators_1.0.14           </span></span>
<span><span class="co">##  [31] pkgconfig_2.0.3             rsvd_1.0.5                 </span></span>
<span><span class="co">##  [33] Matrix_1.7-0                R6_2.5.1                   </span></span>
<span><span class="co">##  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">##  [37] clue_0.3-65                 digest_0.6.35              </span></span>
<span><span class="co">##  [39] colorspace_2.1-0            singscore_1.24.0           </span></span>
<span><span class="co">##  [41] AnnotationDbi_1.66.0        irlba_2.3.5.1              </span></span>
<span><span class="co">##  [43] textshaping_0.3.7           crosstalk_1.2.1            </span></span>
<span><span class="co">##  [45] RSQLite_2.3.6               beachmat_2.20.0            </span></span>
<span><span class="co">##  [47] labeling_0.4.3              gdata_3.0.0                </span></span>
<span><span class="co">##  [49] fansi_1.0.6                 httr_1.4.7                 </span></span>
<span><span class="co">##  [51] abind_1.4-5                 compiler_4.4.0             </span></span>
<span><span class="co">##  [53] bit64_4.0.5                 withr_3.0.0                </span></span>
<span><span class="co">##  [55] doParallel_1.0.17           ROCit_2.1.1                </span></span>
<span><span class="co">##  [57] BiocParallel_1.38.0         DBI_1.2.2                  </span></span>
<span><span class="co">##  [59] highr_0.10                  HDF5Array_1.32.0           </span></span>
<span><span class="co">##  [61] DelayedArray_0.30.0         rjson_0.2.21               </span></span>
<span><span class="co">##  [63] gtools_3.9.5                tools_4.4.0                </span></span>
<span><span class="co">##  [65] glue_1.7.0                  rhdf5filters_1.16.0        </span></span>
<span><span class="co">##  [67] grid_4.4.0                  cluster_2.1.6              </span></span>
<span><span class="co">##  [69] reshape2_1.4.4              generics_0.1.3             </span></span>
<span><span class="co">##  [71] gtable_0.3.5                tidyr_1.3.1                </span></span>
<span><span class="co">##  [73] BiocSingular_1.20.0         ScaledMatrix_1.12.0        </span></span>
<span><span class="co">##  [75] utf8_1.2.4                  XVector_0.44.0             </span></span>
<span><span class="co">##  [77] foreach_1.5.2               pillar_1.9.0               </span></span>
<span><span class="co">##  [79] stringr_1.5.1               GSVA_1.52.0                </span></span>
<span><span class="co">##  [81] limma_3.60.0                circlize_0.4.16            </span></span>
<span><span class="co">##  [83] dplyr_1.1.4                 lattice_0.22-6             </span></span>
<span><span class="co">##  [85] bit_4.0.5                   annotate_1.82.0            </span></span>
<span><span class="co">##  [87] tidyselect_1.2.1            ComplexHeatmap_2.20.0      </span></span>
<span><span class="co">##  [89] SingleCellExperiment_1.26.0 locfit_1.5-9.9             </span></span>
<span><span class="co">##  [91] Biostrings_2.72.0           knitr_1.46                 </span></span>
<span><span class="co">##  [93] bookdown_0.39               edgeR_4.2.0                </span></span>
<span><span class="co">##  [95] xfun_0.43                   statmod_1.5.0              </span></span>
<span><span class="co">##  [97] DT_0.33                     stringi_1.8.3              </span></span>
<span><span class="co">##  [99] UCSC.utils_1.0.0            yaml_2.3.8                 </span></span>
<span><span class="co">## [101] evaluate_0.23               codetools_0.2-20           </span></span>
<span><span class="co">## [103] tibble_3.2.1                BiocManager_1.30.22        </span></span>
<span><span class="co">## [105] graph_1.82.0                cli_3.6.2                  </span></span>
<span><span class="co">## [107] xtable_1.8-4                systemfonts_1.0.6          </span></span>
<span><span class="co">## [109] munsell_0.5.1               jquerylib_0.1.4            </span></span>
<span><span class="co">## [111] Rcpp_1.0.12                 png_0.1-8                  </span></span>
<span><span class="co">## [113] XML_3.99-0.16.1             parallel_4.4.0             </span></span>
<span><span class="co">## [115] pkgdown_2.0.9               ggplot2_3.5.1              </span></span>
<span><span class="co">## [117] blob_1.2.4                  sparseMatrixStats_1.16.0   </span></span>
<span><span class="co">## [119] SpatialExperiment_1.14.0    GSEABase_1.66.0            </span></span>
<span><span class="co">## [121] scales_1.3.0                purrr_1.0.2                </span></span>
<span><span class="co">## [123] crayon_1.5.2                GetoptLong_1.0.5           </span></span>
<span><span class="co">## [125] rlang_1.1.3                 KEGGREST_1.44.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey R. Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
