<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run TB gene signature profiling. — runTBsigProfiler • TBSignatureProfiler</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run TB gene signature profiling. — runTBsigProfiler" />
<meta property="og:description" content="Using some subset of the signatures listed in TBsignatures and
specified scoring algorithms, this function runs gene signature profiling
on an input gene expression dataset. It allows for scores to be computed for
these signatures which can be compared using various visualization tools also
provided in the TBSignatureProfiler package." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TBSignatureProfiler</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/rmd/TBSig_Vignette.html">Command Line Analysis</a>
</li>
<li>
  <a href="../articles/rmd/OriginalModelTutorial.html">Original Models</a>
</li>
<li>
  <a href="../articles/rmd/shinytutorial.html">Interactive Shiny Analysis</a>
</li>
<li>
  <a href="../articles/rmd/SignatureAddition.html">Adding Signatures to the Profiler</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/TBSignatureProfiler">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run TB gene signature profiling.</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/TBSignatureProfiler/blob/master/R/profile.R'><code>R/profile.R</code></a></small>
    <div class="hidden name"><code>runTBsigProfiler.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Using some subset of the signatures listed in <code>TBsignatures</code> and
specified scoring algorithms, this function runs gene signature profiling
on an input gene expression dataset. It allows for scores to be computed for
these signatures which can be compared using various visualization tools also
provided in the TBSignatureProfiler package.</p>
    </div>

    <pre class="usage"><span class='fu'>runTBsigProfiler</span><span class='op'>(</span>
  <span class='va'>input</span>,
  useAssay <span class='op'>=</span> <span class='cn'>NULL</span>,
  signatures <span class='op'>=</span> <span class='cn'>NULL</span>,
  algorithm <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"GSVA"</span>, <span class='st'>"ssGSEA"</span>, <span class='st'>"ASSIGN"</span>, <span class='st'>"PLAGE"</span>, <span class='st'>"Zscore"</span>, <span class='st'>"singscore"</span><span class='op'>)</span>,
  combineSigAndAlgorithm <span class='op'>=</span> <span class='cn'>FALSE</span>,
  assignDir <span class='op'>=</span> <span class='cn'>NULL</span>,
  outputFormat <span class='op'>=</span> <span class='cn'>NULL</span>,
  parallel.sz <span class='op'>=</span> <span class='fl'>0</span>,
  ASSIGNiter <span class='op'>=</span> <span class='fl'>1e+05</span>,
  ASSIGNburnin <span class='op'>=</span> <span class='fl'>50000</span>,
  ssgsea_norm <span class='op'>=</span> <span class='cn'>TRUE</span>,
  update_genes <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>input</th>
      <td><p>an input data object of the class <code>SummarizedExperiment</code>,
<code>data.frame</code>, or <code>matrix</code> containing gene expression data. Required.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>a character string specifying the assay to use for signature
profiling when <code>input</code> is a <code>SummarizedExperiment</code>. Required only for
input data of the class <code>SummarizedExperiment</code>. If null, the assay
used will be <code>"counts"</code>. The default is <code>NULL.</code></p></td>
    </tr>
    <tr>
      <th>signatures</th>
      <td><p>a <code>list</code> of signatures to run with their associated genes.
This list should be in the same format as <code>TBsignatures</code>, included in
the TBSignatureProfiler package. If <code>signatures = NULL</code>, the default set
of signatures <code>TBsignatures</code> list is used. For details, run
<code><a href='TBsignatures.html'>?TBsignatures</a></code>. The default is <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>algorithm</th>
      <td><p>a vector of algorithms to run, or character string if only
one is desired. The default is <code><a href='https://rdrr.io/r/base/c.html'>c("GSVA", "ssGSEA", "ASSIGN",
"PLAGE", "Zscore", "singscore")</a></code>.</p></td>
    </tr>
    <tr>
      <th>combineSigAndAlgorithm</th>
      <td><p>logical, if <code>TRUE</code>, output row names will
be of the form &lt;algorithm&gt;_&lt;signature&gt;. It must be set to codeTRUE if the
<code>ouputFormat</code> will be a SummarizedExperiment and
<code>length(algorithm) &gt; 1</code>.
It will always be <code>FALSE</code> if only one algorithm is selected.
If <code>FALSE</code>, there will be a column named algorithm' that lists which
algorithm is used, and a column named 'pathway' that lists the signature
profiled. If <code>NULL</code>, and one algorithm was used, the algorithm will not
be listed. The default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>assignDir</th>
      <td><p>a character string naming a directory to save intermediate
ASSIGN results if <code>algorithm</code> specifies <code>"ASSIGN"</code>. The default is
<code>NULL</code>, in which case intermediate results will not be saved.</p></td>
    </tr>
    <tr>
      <th>outputFormat</th>
      <td><p>a character string specifying the output data format.
Possible values are <code>"SummarizedExperiment"</code>, <code>"matrix"</code>, or
<code>"data.frame"</code>. The default is to return the same type as the
<code>input</code> object.</p></td>
    </tr>
    <tr>
      <th>parallel.sz</th>
      <td><p>an integer identifying the number of processors to use
when running the calculations in parallel for the GSVA and ssGSEA algorithms.
If <code>parallel.sz = 0</code>, all cores are used. The default is <code>0</code>.</p></td>
    </tr>
    <tr>
      <th>ASSIGNiter</th>
      <td><p>an integer indicating the number of iterations to use in
the MCMC for the ASSIGN algorithm. The default is <code>100,000</code>.</p></td>
    </tr>
    <tr>
      <th>ASSIGNburnin</th>
      <td><p>an integer indicating the number of burn-in iterations
to use in the MCMC for the ASSIGN algorithm. These iterations are discarded
when computing the posterior means of the model parameters. The default is
<code>50,000</code>.</p></td>
    </tr>
    <tr>
      <th>ssgsea_norm</th>
      <td><p>logical, passed to <code><a href='https://rdrr.io/pkg/GSVA/man/gsva.html'>GSVA::gsva()</a></code>. When parameter
<code>algorithm = "ssgsea"</code>,the profiler runs the SSGSEA method from
Barbie et al. (2009) normalizing the scores by the absolute difference
between the minimum and the maximum, as described in their paper.
When <code>ssgsea.norm = FALSE</code>, this last normalization step is skipped.
The default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>update_genes</th>
      <td><p>logical, denotes whether gene names from <code>signatures</code>
and the rownames of <code>input</code> should be checked for accuracy using
<code><a href='https://rdrr.io/pkg/HGNChelper/man/checkGeneSymbols.html'>HGNChelper::checkGeneSymbols()</a></code>. The mapping assumes
genes are from humans and will keep unmapped genes as the original
input gene name. Default is <code>TRUE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p>Profiling for the Z-Score, PLAGE, GSVA, ssGSEA algorithms are all
conducted with the Bioconductor <code>GSVA</code> package. Profiling for the
singscore algorithm is conducted with the Bioconductor <code>singscore</code>
package.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>SummarizedExperiment</code> object, <code>data.frame</code>, or
<code>matrix</code> of signature profiling results. The returned object will be
of the format specified in <code>outputFormat</code>.
If <code>input</code> is a <code>SummarizedExperiment</code> and
<code>outputFormat = "SummarizedExperiment"</code>, then the output will retain
any input information stored in the input colData. In general, if
<code>outputFormat = "SummarizedExperiment"</code> then columns in the <code>colData</code>
will include the scores for each desired signature with samples on the rows.
If <code>input</code> is a <code>data.frame</code> or <code>matrix</code>, then the returned
object will have signatures on the rows and samples on the columns.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Barbie, D.A., Tamayo, P., Boehm, J.S., Kim, S.Y., Moody, S.E., Dunn, I.F., Schinzel, A.C.,
Sandy, P., Meylan, E., Scholl, C., et al. (2009). Systematic RNA interference reveals
that oncogenic KRAS-driven cancers require TBK1. <em>Nature</em> <strong>462</strong>, 108-112.
doi: <a href='https://doi.org/10.1038/nature08460'>10.1038/nature08460</a>.</p>
<p>Foroutan, M. et al. (2018). Single sample scoring of molecular phenotypes.
<em>BMC Bioinformatics</em>, <b>19</b>. doi:
<a href='https://doi.org/10.1186/s12859-018-2435-4'>10.1186/s12859-018-2435-4</a>.</p>
<p>Lee, E. et al. (2008). Inferring pathway activity toward precise disease
classification. <em>PLoS Comp Biol</em>, 4(11):e1000217. doi:
<a href='https://doi.org/10.1371/journal.pcbi.1000217'>10.1371/journal.pcbi.1000217</a></p>
<p>Shen, Y. et al. (2015). ASSIGN: context-specific genomic profiling of
multiple heterogeneous biological pathways. <em>Bioinformatics</em>, <b>31</b>,
1745-1753. doi:
<a href='https://doi.org/10.1093/bioinformatics/btv031'>10.1093/bioinformatics/btv031</a>.</p>
<p>Subramanian, A. et al. (2005). Gene set enrichment analysis: A knowledge-based
approach for interpreting genome-wide expression profiles. <em>PNAS</em>,
<strong>102</strong>, 15545-15550. doi:
<a href='https://doi.org/10.1073/pnas.0506580102'>10.1073/pnas.0506580102</a>.</p>
<p>Tomfohr, J. et al. (2005). Pathway level analysis of gene expression using
singular value decomposition. <em>BMC Bioinformatics</em>, 6:225. doi:
<a href='https://doi.org/10.1186/1471-2105-6-225'>10.1186/1471-2105-6-225</a></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Using a data.frame input/output</span>
 <span class='co'># Create some toy data to test Zak_RISK_16 signature, using 5 samples with low</span>
 <span class='co'># expression &amp; five samples with high expression of the signatures genes.</span>
<span class='va'>df_testdata</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/cbind.html'>rbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fl'>80</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fl'>80</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>5</span><span class='op'>)</span>, <span class='fl'>16</span>, <span class='fl'>10</span>,
                             dimnames <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>TBsignatures</span><span class='op'>$</span><span class='va'>Zak_RISK_16</span>,
                             <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"sample"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,
                      <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fl'>1000</span><span class='op'>)</span>, <span class='fl'>100</span>, <span class='fl'>10</span>,
                             dimnames <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"gene"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span>,
                             <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"sample"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>res</span> <span class='op'>&lt;-</span> <span class='fu'>runTBsigProfiler</span><span class='op'>(</span>input <span class='op'>=</span> <span class='va'>df_testdata</span>,
                        signatures <span class='op'>=</span> <span class='va'>TBsignatures</span><span class='op'>[</span><span class='st'>"Zak_RISK_16"</span><span class='op'>]</span>,
                        algorithm <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"GSVA"</span>, <span class='st'>"ssGSEA"</span><span class='op'>)</span>,
                        combineSigAndAlgorithm <span class='op'>=</span> <span class='cn'>FALSE</span>,
                        parallel.sz <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Parameter update_genes is TRUE. Gene names will be updated.</span></div><div class='output co'>#&gt; <span class='message'>Running GSVA</span></div><div class='output co'>#&gt; Estimating GSVA scores for 1 gene sets.
#&gt; Estimating ECDFs with Gaussian kernels
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%
#&gt; </div><div class='output co'>#&gt; <span class='message'>Running ssGSEA</span></div><div class='output co'>#&gt; Estimating ssGSEA scores for 1 gene sets.
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=======                                                               |  10%  |                                                                              |==============                                                        |  20%  |                                                                              |=====================                                                 |  30%  |                                                                              |============================                                          |  40%  |                                                                              |===================================                                   |  50%  |                                                                              |==========================================                            |  60%  |                                                                              |=================================================                     |  70%  |                                                                              |========================================================              |  80%  |                                                                              |===============================================================       |  90%  |                                                                              |======================================================================| 100%
#&gt; </div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span><span class='op'>(</span><span class='va'>res</span>, <span class='va'>res</span><span class='op'>$</span><span class='va'>pathway</span> <span class='op'>==</span> <span class='st'>"Zak_RISK_16"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;       pathway algorithm            sample1            sample2           sample3
#&gt; 1 Zak_RISK_16      GSVA -0.696424361493125 -0.621188118811882 -0.35357142857143
#&gt; 2 Zak_RISK_16    ssGSEA 0.0527834730744937 0.0433081020509083 0.253310544979381
#&gt;              sample4            sample5           sample6           sample7
#&gt; 1 -0.549418960244649 -0.690000000000001 0.611176470588235              0.68
#&gt; 2  0.461319772273108 -0.112447976848108 0.886480120752086 0.887552023151892
#&gt;             sample8           sample9          sample10
#&gt; 1 0.662450592885375 0.526363636363636 0.606936936936937
#&gt; 2 0.887552023151892 0.887552023151892 0.887552023151892</div><div class='input'>
<span class='co'>## Using a SummarizedExperiment input/output</span>
 <span class='co'># The TB_indian SummarizedExperiment data is included in the package.</span>
<span class='va'>GSVA_res</span> <span class='op'>&lt;-</span> <span class='fu'>runTBsigProfiler</span><span class='op'>(</span>input <span class='op'>=</span> <span class='va'>TB_indian</span>,
                             useAssay <span class='op'>=</span> <span class='st'>"logcounts"</span>,
                             signatures <span class='op'>=</span> <span class='va'>TBsignatures</span><span class='op'>[</span><span class='st'>"Zak_RISK_16"</span><span class='op'>]</span>,
                             algorithm <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"GSVA"</span><span class='op'>)</span>,
                             combineSigAndAlgorithm <span class='op'>=</span> <span class='cn'>FALSE</span>,
                             parallel.sz <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Parameter update_genes is TRUE. Gene names will be updated.</span></div><div class='output co'>#&gt; <span class='message'>Running GSVA</span></div><div class='output co'>#&gt; <span class='warning'>Warning: 1 genes with constant expression values throuhgout the samples.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Since argument method!="ssgsea", genes with constant expression values are discarded.</span></div><div class='output co'>#&gt; Estimating GSVA scores for 1 gene sets.
#&gt; Estimating ECDFs with Gaussian kernels
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%
#&gt; </div><div class='input'><span class='va'>GSVA_res</span><span class='op'>$</span><span class='va'>Zak_RISK_16</span>
</div><div class='output co'>#&gt;  [1]  0.6255317  0.6709665  0.4502341  0.7240338  0.5423896  0.5076658
#&gt;  [7]  0.6268138 -0.3785315  0.5837650  0.3315927  0.5688302  0.5667150
#&gt; [13]  0.6943893 -0.6998614  0.3244909  0.6782273 -0.6715210 -0.6758541
#&gt; [19] -0.5738776 -0.7008174 -0.5409373 -0.5289045 -0.5812383 -0.6514118
#&gt; [25] -0.6726977 -0.2904185 -0.5783098 -0.6327992 -0.6770838 -0.4487168
#&gt; [31] -0.6444862 -0.1045107  0.6871138  0.1003890 -0.4337305 -0.5566492
#&gt; [37]  0.3533240 -0.3596061  0.6574704 -0.3224089  0.2424746  0.2679882
#&gt; [43]  0.1508951  0.3123478</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Aubrey Odom, Xutao Wang, Yue Zhao, Christian Love, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


