<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Using some subset of the signatures listed in TBsignatures and
specified scoring algorithms, this function runs gene signature profiling
on an input gene expression dataset. It allows for scores to be computed for
these signatures which can be compared using various visualization tools also
provided in the TBSignatureProfiler package."><title>Run TB gene signature profiling. — runTBsigProfiler • TBSignatureProfiler</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run TB gene signature profiling. — runTBsigProfiler"><meta property="og:description" content="Using some subset of the signatures listed in TBsignatures and
specified scoring algorithms, this function runs gene signature profiling
on an input gene expression dataset. It allows for scores to be computed for
these signatures which can be compared using various visualization tools also
provided in the TBSignatureProfiler package."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TBSignatureProfiler</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.13.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/rmd/TBSig_Vignette.html">Command Line Analysis</a>
    <a class="dropdown-item" href="../articles/rmd/OriginalModelTutorial.html">Original Models</a>
    <a class="dropdown-item" href="../articles/rmd/shinytutorial.html">Interactive Shiny Analysis</a>
    <a class="dropdown-item" href="../articles/rmd/SignatureAddition.html">Adding Signatures to the Profiler</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">TBSP News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/TBSignatureProfiler">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Run TB gene signature profiling.</h1>
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/TBSignatureProfiler/blob/HEAD/R/profile.R" class="external-link"><code>R/profile.R</code></a></small>
      <div class="d-none name"><code>runTBsigProfiler.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Using some subset of the signatures listed in <code>TBsignatures</code> and
specified scoring algorithms, this function runs gene signature profiling
on an input gene expression dataset. It allows for scores to be computed for
these signatures which can be compared using various visualization tools also
provided in the TBSignatureProfiler package.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">runTBsigProfiler</span><span class="op">(</span></span>
<span>  <span class="va">input</span>,</span>
<span>  useAssay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  signatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span>, <span class="st">"ASSIGN"</span>, <span class="st">"PLAGE"</span>, <span class="st">"Zscore"</span>, <span class="st">"singscore"</span><span class="op">)</span>,</span>
<span>  combineSigAndAlgorithm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  assignDir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  outputFormat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  parallel.sz <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  ASSIGNiter <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  ASSIGNburnin <span class="op">=</span> <span class="fl">50000</span>,</span>
<span>  ssgsea_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  update_genes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="source">Source<a class="anchor" aria-label="anchor" href="#source"></a></h2>
    <p>Profiling for the Z-Score, PLAGE, GSVA, ssGSEA algorithms are all
conducted with the Bioconductor <code>GSVA</code> package. Profiling for the
singscore algorithm is conducted with the Bioconductor <code>singscore</code>
package.</p>
    </div>
    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>input</dt>
<dd><p>an input data object of the class <code>SummarizedExperiment</code>,
<code>data.frame</code>, or <code>matrix</code> containing gene expression data. Required.</p></dd>


<dt>useAssay</dt>
<dd><p>a character string specifying the assay to use for signature
profiling when <code>input</code> is a <code>SummarizedExperiment</code>. Required only for
input data of the class <code>SummarizedExperiment</code>. If null, the assay
used will be <code>"counts"</code>. The default is <code>NULL.</code></p></dd>


<dt>signatures</dt>
<dd><p>a <code>list</code> of signatures to run with their associated genes.
This list should be in the same format as <code>TBsignatures</code>, included in
the TBSignatureProfiler package. If <code>signatures = NULL</code>, the default set
of signatures <code>TBsignatures</code> list is used.
For details, run <code><a href="TBsignatures.html">?TBsignatures</a></code>.
If &lt;2 genes in a signature are present in the sample, that signature will
not be evaluated and will not be present in the resulting SE object.
The default is <code>NULL</code>.</p></dd>


<dt>algorithm</dt>
<dd><p>a vector of algorithms to run, or character string if only
one is desired. The default is <code>c("GSVA", "ssGSEA", "ASSIGN",
"PLAGE", "Zscore", "singscore")</code>.</p></dd>


<dt>combineSigAndAlgorithm</dt>
<dd><p>logical, if <code>TRUE</code>, output row names will
be of the form &lt;algorithm&gt;_&lt;signature&gt;. It must be set to codeTRUE if the
<code>ouputFormat</code> will be a SummarizedExperiment and
<code>length(algorithm) &gt; 1</code>.
It will always be <code>FALSE</code> if only one algorithm is selected.
If <code>FALSE</code>, there will be a column named algorithm' that lists which
algorithm is used, and a column named 'pathway' that lists the signature
profiled. If <code>NULL</code>, and one algorithm was used, the algorithm will not
be listed. The default is <code>FALSE</code>.</p></dd>


<dt>assignDir</dt>
<dd><p>a character string naming a directory to save intermediate
ASSIGN results if <code>algorithm</code> specifies <code>"ASSIGN"</code>. The default is
<code>NULL</code>, in which case intermediate results will not be saved.</p></dd>


<dt>outputFormat</dt>
<dd><p>a character string specifying the output data format.
Possible values are <code>"SummarizedExperiment"</code>, <code>"matrix"</code>, or
<code>"data.frame"</code>. The default is to return the same type as the
<code>input</code> object.</p></dd>


<dt>parallel.sz</dt>
<dd><p>an integer identifying the number of processors to use
when running the calculations in parallel for the GSVA and ssGSEA algorithms.
If <code>parallel.sz = 0</code>, all cores are used. The default is <code>0</code>.</p></dd>


<dt>ASSIGNiter</dt>
<dd><p>an integer indicating the number of iterations to use in
the MCMC for the ASSIGN algorithm. The default is <code>100,000</code>.</p></dd>


<dt>ASSIGNburnin</dt>
<dd><p>an integer indicating the number of burn-in iterations
to use in the MCMC for the ASSIGN algorithm. These iterations are discarded
when computing the posterior means of the model parameters. The default is
<code>50,000</code>.</p></dd>


<dt>ssgsea_norm</dt>
<dd><p>logical, passed to <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">GSVA::gsva()</a></code>. When parameter
<code>algorithm = "ssgsea"</code>,the profiler runs the SSGSEA method from
Barbie et al. (2009) normalizing the scores by the absolute difference
between the minimum and the maximum, as described in their paper.
When <code>ssgsea.norm = FALSE</code>, this last normalization step is skipped.
The default is <code>TRUE</code>.</p></dd>


<dt>update_genes</dt>
<dd><p>logical, denotes whether gene names from <code>signatures</code>
and the rownames of <code>input</code> should be checked for accuracy using
<code><a href="https://rdrr.io/pkg/HGNChelper/man/checkGeneSymbols.html" class="external-link">HGNChelper::checkGeneSymbols()</a></code>. The mapping assumes
genes are from humans and will keep unmapped genes as the original
input gene name. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code>SummarizedExperiment</code> object, <code>data.frame</code>, or
<code>matrix</code> of signature profiling results. The returned object will be
of the format specified in <code>outputFormat</code>.
If <code>input</code> is a <code>SummarizedExperiment</code> and
<code>outputFormat = "SummarizedExperiment"</code>, then the output will retain
any input information stored in the input colData. In general, if
<code>outputFormat = "SummarizedExperiment"</code> then columns in the <code>colData</code></p>


<p>will include the scores for each desired signature with samples on the rows.
If <code>input</code> is a <code>data.frame</code> or <code>matrix</code>, then the returned
object will have signatures on the rows and samples on the columns.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Barbie, D.A., Tamayo, P., Boehm, J.S., Kim, S.Y., Moody, S.E., Dunn, I.F., Schinzel, A.C.,
Sandy, P., Meylan, E., Scholl, C., et al. (2009). Systematic RNA interference reveals
that oncogenic KRAS-driven cancers require TBK1. <em>Nature</em> <strong>462</strong>, 108-112.
doi: <a href="https://doi.org/10.1038/nature08460" class="external-link">10.1038/nature08460</a>.</p>
<p>Foroutan, M. et al. (2018). Single sample scoring of molecular phenotypes.
<em>BMC Bioinformatics</em>, <b>19</b>. doi:
<a href="https://doi.org/10.1186/s12859-018-2435-4" class="external-link">10.1186/s12859-018-2435-4</a>.</p>
<p>Lee, E. et al. (2008). Inferring pathway activity toward precise disease
classification. <em>PLoS Comp Biol</em>, 4(11):e1000217. doi:
<a href="https://doi.org/10.1371/journal.pcbi.1000217" class="external-link">10.1371/journal.pcbi.1000217</a></p>
<p>Shen, Y. et al. (2015). ASSIGN: context-specific genomic profiling of
multiple heterogeneous biological pathways. <em>Bioinformatics</em>, <b>31</b>,
1745-1753. doi:
<a href="https://doi.org/10.1093/bioinformatics/btv031" class="external-link">10.1093/bioinformatics/btv031</a>.</p>
<p>Subramanian, A. et al. (2005). Gene set enrichment analysis: A knowledge-based
approach for interpreting genome-wide expression profiles. <em>PNAS</em>,
<strong>102</strong>, 15545-15550. doi:
<a href="https://doi.org/10.1073/pnas.0506580102" class="external-link">10.1073/pnas.0506580102</a>.</p>
<p>Tomfohr, J. et al. (2005). Pathway level analysis of gene expression using
singular value decomposition. <em>BMC Bioinformatics</em>, 6:225. doi:
<a href="https://doi.org/10.1186/1471-2105-6-225" class="external-link">10.1186/1471-2105-6-225</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Using a data.frame input/output</span></span></span>
<span class="r-in"><span> <span class="co"># Create some toy data to test Zak_RISK_16 signature, using 5 samples with low</span></span></span>
<span class="r-in"><span> <span class="co"># expression &amp; five samples with high expression of the signatures genes.</span></span></span>
<span class="r-in"><span><span class="va">df_testdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">80</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">80</span><span class="op">)</span> <span class="op">+</span> <span class="fl">5</span><span class="op">)</span>, <span class="fl">16</span>, <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                             dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">TBsignatures</span><span class="op">$</span><span class="va">Zak_RISK_16</span>,</span></span>
<span class="r-in"><span>                             <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, <span class="fl">100</span>, <span class="fl">10</span>,</span></span>
<span class="r-in"><span>                             dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                             <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">runTBsigProfiler</span><span class="op">(</span>input <span class="op">=</span> <span class="va">df_testdata</span>,</span></span>
<span class="r-in"><span>                        signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Zak_RISK_16"</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                        algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSVA"</span>, <span class="st">"ssGSEA"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                        combineSigAndAlgorithm <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                        parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Parameter update_genes is TRUE. Gene names will be updated.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running GSVA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating GSVA scores for 1 gene sets.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating ECDFs with Gaussian kernels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running ssGSEA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating ssGSEA scores for 1 gene sets.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Calculating ranks..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Calculating absolute values from ranks..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=======                                                               |  10%  |                                                                              |==============                                                        |  20%  |                                                                              |=====================                                                 |  30%  |                                                                              |============================                                          |  40%  |                                                                              |===================================                                   |  50%  |                                                                              |==========================================                            |  60%  |                                                                              |=================================================                     |  70%  |                                                                              |========================================================              |  80%  |                                                                              |===============================================================       |  90%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Normalizing..."</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">res</span><span class="op">$</span><span class="va">pathway</span> <span class="op">==</span> <span class="st">"Zak_RISK_16"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       pathway algorithm            sample1            sample2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 Zak_RISK_16      GSVA -0.583595505617978  -0.66233009708738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 Zak_RISK_16    ssGSEA   0.04206119542406 -0.160101597476311</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              sample3             sample4            sample5           sample6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 -0.550000000000001  -0.625470085470087 -0.550000000000001 0.630731707317073</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.0710783789344759 0.00506292026301021  0.214919315526654 0.839898402523689</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sample7           sample8           sample9          sample10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.541536523929471 0.617407407407407 0.618888888888889 0.545531335149864</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.839898402523689 0.839898402523689 0.839898402523689 0.839898402523689</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Using a SummarizedExperiment input/output</span></span></span>
<span class="r-in"><span> <span class="co"># The TB_indian SummarizedExperiment data is included in the package.</span></span></span>
<span class="r-in"><span><span class="va">GSVA_res</span> <span class="op">&lt;-</span> <span class="fu">runTBsigProfiler</span><span class="op">(</span>input <span class="op">=</span> <span class="va">TB_indian</span>,</span></span>
<span class="r-in"><span>                             useAssay <span class="op">=</span> <span class="st">"logcounts"</span>,</span></span>
<span class="r-in"><span>                             signatures <span class="op">=</span> <span class="va">TBsignatures</span><span class="op">[</span><span class="st">"Zak_RISK_16"</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                             algorithm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSVA"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                             combineSigAndAlgorithm <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                             parallel.sz <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Parameter update_genes is TRUE. Gene names will be updated.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running GSVA</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>1 genes with constant expression values throuhgout the samples.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Since argument method!="ssgsea", genes with constant expression values are discarded.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating GSVA scores for 1 gene sets.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating ECDFs with Gaussian kernels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">GSVA_res</span><span class="op">$</span><span class="va">Zak_RISK_16</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1]  0.6255317  0.6709665  0.4502341  0.7240338  0.5423896  0.5076658</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7]  0.6268138 -0.3785315  0.5837650  0.3315927  0.5688302  0.5667150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13]  0.6943893 -0.6998614  0.3244909  0.6782273 -0.6715210 -0.6758541</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] -0.5738776 -0.7008174 -0.5409373 -0.5289045 -0.5812383 -0.6514118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] -0.6726977 -0.2904185 -0.5783098 -0.6327992 -0.6770838 -0.4487168</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] -0.6444862 -0.1045107  0.6871138  0.1003890 -0.4337305 -0.5566492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [37]  0.3533240 -0.3596061  0.6574704 -0.3224089  0.2424746  0.2679882</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [43]  0.1508951  0.3123478</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Site created by Aubrey R. Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

